frontend/libs/studio-components/src/components/StudioTextResourcePicker/StudioTextResourcePicker.test.tsx

```diff
@@ -58,19 +60,35 @@ describe('StudioTextResourcePicker', () => {
     expect(onValueChange).toHaveBeenCalledWith(textResourceToPick.id);
   });
 
-  it('Displays the empty list text when the user clicks and there are no text resources', async () => {
-    const user = userEvent.setup();
-    renderTextResourcePicker({ textResources: [] });
-    await user.click(getCombobox());
-    expect(screen.getByText(emptyListText)).toBeInTheDocument();
-  });
-
   it("Renders with the text of the text resource of which the ID is given by the component's value prop", () => {
     const pickedTextResource = textResources[129];
     renderTextResourcePicker({ value: pickedTextResource.id });
     expect(getCombobox()).toHaveValue(pickedTextResource.value);
   });
 
+  it('Displays the unset option when the user clicks', async () => {
+    const user = userEvent.setup();
+    renderTextResourcePicker();
+    await user.click(getCombobox());
+    expect(screen.getByRole('option', { name: unsetOptionLabel })).toBeInTheDocument();
+  });
+
+  it('Renders with the unset option selected by default', () => {
+    renderTextResourcePicker();
+    expect(getCombobox()).toHaveValue(unsetOptionLabel);
+  });
+
+  it('Calls the onValueChange callback with null when the user selects the unset option', async () => {
```

Comment:
Not sure, but would it make sense also testing;
1. that the callBack is called with `null` also when the value is already `unsetOptionLabel`?
2. that the value in the combobox is `unsetOptionLabel` after the click?

Reply:
👍

Reply:
I don't understand why my comment was published in your name ... Maybe I clicked "edit" instead of "reply".